lib/rules/no-extra-parens.js

```diff
@@ -73,6 +75,38 @@ module.exports = {
          * @private
          */
         function ruleApplies(node) {
+            if (node.type === "JSXElement") {
+                const isSelfClosing = node.openingElement.selfClosing;
+
+                // Self-closing components do not have `closingElement`, so we must check the start
+                // and end lines of this particular component directly
+                const isSingleLine = isSelfClosing
+                    ? node.loc.start.line === node.loc.end.line
+                    : astUtils.isTokenOnSameLine(node.openingElement, node.closingElement);
```

Comment:
Ah, of course! I spent quite some time figuring out how to detect multi-line, multi-element JSX structures, and only much later realised that a single-element, multi-line JSX element is possible. I guess I got "thought-boxed". Updated to only use the `loc` info of the current node.